cfd_protocol/src/lib.rs

```diff
@@ -731,95 +745,55 @@ impl CommitTransaction {
-                out.script_pubkey != DUMMY_2OF2_MULITISIG.parse().expect("To be a valid script")
-            })
-            .collect::<Vec<_>>();
-
-        let taker_change = taker_psbt
-            .global
-            .unsigned_tx
-            .output
-            .into_iter()
-            .filter(|out| {
-                out.script_pubkey != DUMMY_2OF2_MULITISIG.parse().expect("To be a valid script")
-            })
-            .collect();
-
-        let lock_output = TxOut {
-            value: amount.as_sat(),
-            script_pubkey: lock_descriptor.script_pubkey(),
-        };
-
-        let lock_tx = Transaction {
-            version: 2,
-            lock_time: 0,
-            input: vec![
-                maker_psbt.global.unsigned_tx.input,
-                taker_psbt.global.unsigned_tx.input,
-            ]
-            .concat(),
-            output: vec![vec![lock_output], maker_change, taker_change].concat(),
-        };
-
-        let inner = PartiallySignedTransaction {
-            global: Global::from_unsigned_tx(lock_tx)?,
-            inputs: vec![maker_psbt.inputs, taker_psbt.inputs].concat(),
-            outputs: vec![maker_psbt.outputs, taker_psbt.outputs].concat(),
-        };
+) -> PartiallySignedTransaction {
+    let lock_descriptor = lock_descriptor(maker_pk, taker_pk);
 
-        Ok(Self {
-            inner,
-            lock_descriptor,
-            amount,
+    let maker_change = maker_psbt
+        .global
+        .unsigned_tx
+        .output
+        .into_iter()
+        .filter(|out| {
+            out.script_pubkey != DUMMY_2OF2_MULITISIG.parse().expect("To be a valid script")
         })
-    }
+        .collect::<Vec<_>>();
 
-    fn lock_outpoint(&self) -> OutPoint {
-        let txid = self.inner.global.unsigned_tx.txid();
-        let vout = self
-            .inner
-            .global
-            .unsigned_tx
-            .output
-            .iter()
-            .position(|out| out.script_pubkey == self.lock_descriptor.script_pubkey())
-            .expect("to find lock output in lock tx");
+    let taker_change = taker_psbt
+        .global
+        .unsigned_tx
+        .output
+        .into_iter()
+        .filter(|out| {
+            out.script_pubkey != DUMMY_2OF2_MULITISIG.parse().expect("To be a valid script")
+        })
+        .collect();
 
-        OutPoint {
-            txid,
-            vout: vout as u32,
-        }
-    }
+    let lock_output = TxOut {
+        value: amount.as_sat(),
+        script_pubkey: lock_descriptor.script_pubkey(),
+    };
 
-    fn descriptor(&self) -> Descriptor<PublicKey> {
-        self.lock_descriptor.clone()
-    }
+    let lock_tx = Transaction {
+        version: 2,
+        lock_time: 0,
+        input: vec![
+            maker_psbt.global.unsigned_tx.input,
+            taker_psbt.global.unsigned_tx.input,
+        ]
+        .concat(),
+        output: vec![vec![lock_output], maker_change, taker_change].concat(),
```

Comment:
I think this double `Vec` is a bit confusing. Instead of `collect`ing all the individual iterators, you could concatenated them and use `iter::once` for the `lock_output`.